gologr-wlog/gologr.go

```diff
@@ -0,0 +1,69 @@
+package gologrwlog
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/go-logr/logr"
+	"github.com/palantir/witchcraft-go-logging/wlog/svclog/svc1log"
+)
+
+type goLogrWrapper struct {
+	origin string
+	logger svc1log.Logger
+}
+
+// NewGoLogrWrapper returns a go-logr Logger implementation that uses svc1log to write logs.
+func NewGoLogrWrapper(logger svc1log.Logger, origin string) logr.Logger {
+	logger = svc1log.WithParams(logger, svc1log.Origin(origin))
+	return &goLogrWrapper{
+		origin: origin,
+		logger: logger,
+	}
+}
+
+func (s *goLogrWrapper) Info(msg string, keysAndValues ...interface{}) {
+	s.logger.Info(msg, toSafeParams(s.logger, keysAndValues))
+}
+
+func (s *goLogrWrapper) Enabled() bool {
+	return true
+}
+
+func (s *goLogrWrapper) Error(err error, msg string, keysAndValues ...interface{}) {
+	s.logger.Error(msg, svc1log.Stacktrace(err), toSafeParams(s.logger, keysAndValues))
+}
+
+func (s *goLogrWrapper) V(level int) logr.InfoLogger {
+	return NewGoLogrWrapper(s.logger, s.origin)
+}
+
+func (s *goLogrWrapper) WithValues(keysAndValues ...interface{}) logr.Logger {
+	logger := svc1log.WithParams(s.logger, toSafeParams(s.logger, keysAndValues))
+	return NewGoLogrWrapper(logger, s.origin)
+}
+
+func (s *goLogrWrapper) WithName(name string) logr.Logger {
+	return NewGoLogrWrapper(s.logger, filepath.Join(s.origin, name))
+}
+
+func toSafeParams(logger svc1log.Logger, keysAndValues []interface{}) svc1log.Param {
+	if len(keysAndValues)%2 == 1 {
+		logger.Error("KeysAndValues pair slice has an odd number of arguments; ignoring all",
```

Comment:
Read this https://dave.cheney.net/2015/11/05/lets-talk-about-logging and decided to bump these up from `Warn` -> `Error`